.github/workflows/pull-request.yml

```diff
@@ -1,4 +1,4 @@
-name: Flutter CI
+name: Pull Request Workflow
```

Comment:
It's on both PR and push. While you are at it, why we don't default only to PR? We don't honestly care if an individual commit breaks anything (or fail tests), we only care for branches and its respective pull requests being made to the `main` anyways.

Reply:
Agreed